File: bin/jwt_generator.ts
#!ts-node

const { sign } = require("jsonwebtoken");
const crypto = require("crypto");
import * as dotenv from "dotenv";

dotenv.config();

const key_name = process.env.COINBASE_API_KEY!;
const key_secret = process.env.COINBASE_PRIVATE_KEY!;

const request_method = "GET";
const url = "api.coinbase.com";
const request_path = "/api/v3/brokerage/accounts";

const algorithm = "HS256";
const uri = request_method + " " + url + request_path;

const token = sign(
  {
    iss: "cdp",
    nbf: Math.floor(Date.now() / 1000),
    exp: Math.floor(Date.now() / 1000) + 120,
    sub: key_name,
    uri,
  },
  key_secret,
  {
    algorithm,
    header: {
      kid: key_name,
      nonce: crypto.randomBytes(16).toString("hex"),
    },
  }
);
console.log(token);

File: scripts/jwt_generator.py
#!python

import os
from coinbase import jwt_generator
from dotenv import load_dotenv

load_dotenv()

API_KEY = os.environ.get("COINBASE_API_KEY")
API_SECRET = os.environ.get("COINBASE_API_SECRET")

request_method = "GET"
request_path = "/api/v3/brokerage/accounts"

def main():
    jwt_uri = jwt_generator.format_jwt_uri(request_method, request_path)
    jwt_token = jwt_generator.build_rest_jwt(jwt_uri, API_KEY, API_SECRET)
    print(jwt_token)

if __name__ == "__main__":
    main()